### PR TITLE
Hotspot/FreeBSD list system processes implementation

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/hotspot/os/bsd/os_perf_bsd.cpp [hunk 1]
```diff
@@ -432,6 +432,58 @@ int SystemProcessInterface::SystemProcesses::system_processes(SystemProcess** sy
...
+    free(lproc);
+    return OS_ERR;
+  }
+
+  pid_count = length / sizeof(*lproc);
+  int process_count = 0;
+  SystemProcess *next;
+  
+  for (int i = 0; i < pid_count; i++) {
+     int bmib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, lproc[i].ki_pid };
```
Comment (battleblow) [inline src/hotspot/os/bsd/os_perf_bsd.cpp]: I'm curious as to what led you to choose "bmib" as the variable name here?  I was thinking "pmib" maybe would be clearer, but I'd like to know the reasoning first :)

File: src/hotspot/os/bsd/os_perf_bsd.cpp [hunk 2]
```diff
@@ -432,6 +432,58 @@ int SystemProcessInterface::SystemProcesses::system_processes(SystemProcess** sy
...
+     length = sizeof(buffer);
+     if (sysctl(bmib, bmiblen, buffer, &length, NULL, 0) == -1) {
+       continue;
+     }
+
+     length = strnlen(buffer, PATH_MAX);
+     if (length > 0) {
+       SystemProcess* current = new SystemProcess();
+       char * path = NEW_C_HEAP_ARRAY(char, length + 1, mtInternal);
+       strcpy(path, buffer);
```
Comment (battleblow) [inline src/hotspot/os/bsd/os_perf_bsd.cpp]: Please avoid the use of strcpy, as it's a potential vector for buffer overflow attacks.  I'd suggest strlcpy or snprintf as possible replacements here.

**Review by battleblow — CHANGES_REQUESTED** [review]

File: src/hotspot/os/bsd/os_perf_bsd.cpp (hunk 2 above)
Comment (devnexen) [reply src/hotspot/os/bsd/os_perf_bsd.cpp]: In general yes but the length is safe but will change it np.

File: src/hotspot/os/bsd/os_perf_bsd.cpp (hunk 1 above)
Comment (devnexen) [reply src/hotspot/os/bsd/os_perf_bsd.cpp]: b as second request but I can change to p no worries 😉

**Comment by battleblow** [comment]
Thanks again for this!

**Pull request merged**